library/src/main/kotlin/org/codice/compliance/SAMLComplianceException.kt

```diff
@@ -10,36 +10,56 @@ import de.jupf.staticlog.Log
 import de.jupf.staticlog.core.LogLevel
 import org.w3c.dom.Node
 
+@Suppress("StringLiteralDuplication")
 class SAMLComplianceException : Exception {
 
-    private constructor(message: String) : super(message)
+    val errorCodes = mutableListOf<SAMLSpecRefMessage>()
```

Comment:
✏️ Instead of making this mutable, could do the following:
```kotlin
    val errorCodes: List<SAMLSpecRefMessage>

    private constructor(codes: Array<out SAMLSpecRefMessage>, message: String) : super(message) {
        errorCodes = listOf(*codes)
    }

    private constructor(codes: Collection<SAMLSpecRefMessage>, message: String) : super(message) {
        errorCodes = codes.toList()
    }

    private constructor(
        codes: Array<out SAMLSpecRefMessage>,
        message: String,
        cause: Throwable
    ) : super(message, cause) {
        errorCodes = listOf(*codes)
    }

    private constructor(
        codes: Collection<SAMLSpecRefMessage>,
        message: String,
        cause: Throwable
    ) : super(message, cause) {
        errorCodes = codes.toList()
    }
```